plutus-core/plutus-core/test/CBOR/DataStability.hs

```diff
@@ -0,0 +1,126 @@
+-- misleading places.
+maketest :: (String, Text, Text, Data) -> TestTree
+maketest (name, s1, s2, expected) =
+    testCaseSteps name $ \step -> do
+      let !bs1 = BSL.fromStrict . fromJust . decodeHex $ s1
+          size1 = BSL.length bs1
+      step $ printf "Deserialising Data object 1 from CBOR bytestring (%s)" (numBytes size1)
+      let !object1 = deserialise @Data bs1
+      let !bs2 = BSL.fromStrict . fromJust . decodeHex $ s2
+          size2 = BSL.length bs2
+      step $ printf "Deserialising Data object 2 from CBOR bytestring (%s)" (numBytes size2)
+      let !object2 = deserialise @Data bs2
+      step $ printf "CBOR size difference: %s" (numBytes (size2 - size1))
+      step "Checking that object1 decoded correctly"
+      object1 @?= expected
+      step "Checking that object2 decoded correctly"
+      object2 @?= expected
+      step "Checking object1 and object2 for equality"
+      assertEqual "Deserialisation produced different objects" object1 object2
+          where numBytes s =
+                    if s == 1
+                    then "1 byte"
+                    else printf "%d bytes" s :: String
+
+-- A list of tuples containing (Script/data-item name, encoded version 1,
+-- encoded version 2, expected object)
+testData :: [(String, Text, Text, Data)]
+testData =
+    [
+     ( "auction_1-1"
+     , "d87982581cbd99a373075d42fe4ac9109515e46303d0940cb9620bf058b87986a9d87980"
+     , "d8799f581cbd99a373075d42fe4ac9109515e46303d0940cb9620bf058b87986a9d87980ff"
+     , Constr 0 [B "\189\153\163s\a]B\254J\201\DLE\149\NAK\228c\ETX\208\148\f\185b\v\240X\184y\134\169", Constr 0 []]
+     )
+    ,
+     ( "future-increase-margin-5 (a)"
+     , "d87981d8798281d879824081d879824019084b81d879824081d87982401908af"
+     , "d8799fd8799f9fd8799f409fd8799f4019084bffffffff9fd8799f409fd8799f401908afffffffffffff"
+     , Constr 0 [Constr 0 [List [Constr 0 [B "",List [Constr 0 [B "",I 2123]]]],List [Constr 0 [B "",List [Constr 0 [B "",I 2223]]]]]]
+     )
+    ,
+     ( "future-increase-margin-5 (b)"
+     , "d87a81d879835840c40f1dc048cc8b8b490cf6f58fbb582f01ecb8199094ed84961ab079dda95df47930e1607f41806587229912a670b64f2c6e67db22c2187781fce00df43c240f5820d8af98eecf2d0c875462713ae861164de002f9a0830f01b19dc33a4e27592513d8798281d879824081d87982401904641b000001739c8a86d8"
+     , "d87a9fd8799f5840c40f1dc048cc8b8b490cf6f58fbb582f01ecb8199094ed84961ab079dda95df47930e1607f41806587229912a670b64f2c6e67db22c2187781fce00df43c240f5820d8af98eecf2d0c875462713ae861164de002f9a0830f01b19dc33a4e27592513d8799f9fd8799f409fd8799f40190464ffffffff1b000001739c8a86d8ffffff"
+     , Constr 1 [Constr 0 [B "\196\SI\GS\192H\204\139\139I\f\246\245\143\187X/\SOH\236\184\EM\144\148\237\132\150\SUB\176y\221\169]\244y0\225`\DELA\128e\135\"\153\DC2\166p\182O,ng\219\"\194\CANw\129\252\224\r\244<$\SI",B "\216\175\152\238\207-\f\135Tbq:\232a\SYNM\224\STX\249\160\131\SI\SOH\177\157\195:N'Y%\DC3",Constr 0 [List [Constr 0 [B "",List [Constr 0 [B "",I 1124]]]],I 1596059191000]]]
```

Comment:
I really wish we could have hex strings in here rather than all this `\196\SI\GS\192H\204\139\139I\f\246....` stuff, but I don't think that's possible.

Reply:
You could put the bytestrings in files and read them in...